File: client/src/app/core/models/tickets/ticket.ts
export interface Ticket {
    id: string;
    raffleId: number;
    ticketNumber: string;
    price: number;
    status: TicketStatus;
    reservationFlag: string;
    reservationTime: Date;  
    customerId: string;
}

export type TicketStatus = 'AVAILABLE' | 'RESERVERD' | 'SOLD';

File: server/services/tickets-service/src/main/java/com/raffleease/tickets_service/Tickets/Models/Ticket.java
package com.raffleease.tickets_service.Tickets.Models;

import com.raffleease.common_models.DTO.Tickets.TicketStatus;
import lombok.*;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@NoArgsConstructor
@AllArgsConstructor
@Data
@Builder
@Document(collection = "Tickets")
public class Ticket {
    @Id
    private String id;
    private Long raffleId;
    private String ticketNumber;
    private BigDecimal price;
    private TicketStatus status;
    private String reservationFlag;
    private LocalDateTime reservationTime;
    private String customerId;
}